hangman.rb

```diff
@@ -0,0 +1,149 @@
+
+class Hangman
+	def initialize
+
+		# Library of words
+		@words = ["orca", "cheese", "cake", "pumpkin", "sun", "apple", "cloud", "paint", "bird", "travel", "lollipop", "ada"]
+
+		@answer_word = choose_word # chooses answer (array of characters)
+		@answer_copy = @answer_word.dup # creates arrage of correct length for @word_display
+		@word_display = @answer_copy.fill("_")
+		@letters_guessed = []
+
+		@default_guesses = 7
+		@wrong_count = 0
+
+		start
+	end
+
+	def start
+		@keep_playing = true
+		puts "Let's play HANGMAN!"
+
+		update_visuals
+		play
+	end
+
+	def play
+		# Checks for win or lose
+		while @keep_playing == true
+			if @wrong_count < 7 && win != true
+				puts "Guess one letter and then hit enter. Type \"quit\" if you don't want to play."
+				print "YOUR GUESS: "
+				@current_guess = get_guess # gets guess from user
+				check_guess(@current_guess) # result of checked guess and updates number of wrong gueses
+			elsif win == true
+				puts "You win! The word was #{@word}! Congrats :)"
+				@keep_playing = false
+			else
+				@keep_playing = false
+				puts "Sorry, you lose. The word was #{@word}."
+				exit
+			end
+		end
+	end
+
+	# get_guess is called in play method
+	def get_guess
+		guess = gets.chomp.downcase.delete(" ")
+
+		if guess == "quit"
+			exit
+		end
+
+		# guess is string
+		return guess
+	end
+
+	# check_guess is called in play method
+	def check_guess(guess)
+
+		if @answer_word.include?(guess) # Correct guess
+
+			@answer_word.each_index do |index| # Checks and replaces correct guesses
+		 		if @answer_word[index] == guess
+		 			@word_display.delete_at(index)
+					@word_display.insert(index, guess)
+				end
+			end
+
+			puts "Yay! You guessed a correct letter."
+
+		else # Incorrect guess
+			puts "Wrong guess, try again! :("
+			@wrong_count += 1
+			puts "Guesses left: #{@default_guesses - @wrong_count}"
+		end
+
+		@letters_guessed.push(guess)
+		update_visuals
+	end
+
+
+	# ANSWER METHODS ---------------------------------------------------------------
+
+	# Chooses the answer from the library
+	def choose_word
+	 	# chooses random word from library
+	 	@word = @words[rand(0...12)]
+	 	# creates array of letters from word
+	 	word_array = @word.split(//)
+		return word_array
+ 	end
+
+	# Checks the guessed word against the answer
+	def win
+		@answer_word == @word_display
+	end
+
+
+	# VISUAL METHODS ---------------------------------------------------------------
+
+	def update_visuals
+		# Prints body and words
+		puts body
```

Comment:
With your more solidified understanding of returns and methods you would probably want to have the `puts` in the `body` method itself rather than using the `puts` here